File: app/src/main/java/com/service/endlessservicewtbroadcastreceiver/utils/ServiceAdmin.java
package com.service.endlessservicewtbroadcastreceiver.utils;


import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;

public class ServiceAdmin {
    private static final String TAG = "ServiceAdmin";
    private static Intent serviceIntent = null;

    public ServiceAdmin() {
    }

    private void setServiceIntent(Context context) {
        if (serviceIntent == null) {
            serviceIntent = new Intent(context, AutoStartService.class);
        }
    }

    public void launchService(Context context) {
        if (context == null) {
            return;
        }
        setServiceIntent(context);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(serviceIntent);
        } else {
            context.startService(serviceIntent);
        }
        Log.d(TAG, "launchService:  Service is starting....");
    }
}

File: app/src/main/java/com/service/endlessservicewtbroadcastreceiver/utils/ServiceAdmin.kt
package com.service.endlessservicewtbroadcastreceiver.utils

import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log

class ServiceAdmin {
    private fun setServiceIntent(context: Context) {
        if (serviceIntent == null) {
            serviceIntent = Intent(context, LocationForegroundtService::class.java)
        }
    }

    fun launchService(context: Context?) {
        context?.let {
            setServiceIntent(it)
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                it.startForegroundService(serviceIntent)
            } else {
                it.startService(serviceIntent)
            }
            Log.d(TAG, "launchService:  Service is starting....")
        }
    }

    companion object {
        private const val TAG = "ServiceAdmin"
        private var serviceIntent: Intent? = null
    }
}